setup/install/providers/azure.md

```diff
@@ -52,16 +52,22 @@ RESOURCE_GROUP="Spinnaker"
 az group create --name $RESOURCE_GROUP --location <Insert Location>
 ```
 
-Finally, create a Key Vault (where the vault name is globally unique) and add a default username/password:
+Finally, create a Key Vault (where the vault name is globally unique) and add a default username/ssh public key. This credential is used to provision Azure VM scale set. You can log on to VM instances in the VM scale set with this credential.
 
 ```bash
 VAULT_NAME=<Insert Vault Name>
 az keyvault create --enabled-for-template-deployment true --resource-group $RESOURCE_GROUP --name $VAULT_NAME
 az keyvault set-policy --secret-permissions get --name $VAULT_NAME --spn $APP_ID
 az keyvault secret set --name VMUsername --vault-name $VAULT_NAME --value <Insert default username>
-az keyvault secret set --name VMPassword --vault-name $VAULT_NAME --value <Insert default password>
+az keyvault secret set --name VMSshPublicKey --vault-name $VAULT_NAME --value <Insert default SSH public key>
 ```
 
+> NOTE: If you prefer to use password instead of SSH public key, then replace<br/>
+`az keyvault secret set --name VMSshPublicKey --vault-name $VAULT_NAME --value` `<Insert default SSH public key>`<br/>
+with<br/>
+`az keyvault secret set --name VMPassword --vault-name $VAULT_NAME --value` `<Insert default password>`<br/>
+Azure has requirements for setting the username and password to provision VM. Follow the [rule](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/faq#what-are-the-username-requirements-when-creating-a-vm){:target="\_blank"} otherwise the provisioning of VM scale set will fail.
```

Comment:
I think you can change this whole paragraph to...
"Follow the password rules documented [here](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/faq#what-are-the-username-requirements-when-creating-a-vm)"

That is, I think it's safe to assume the reader knows the rules apply to provisioning the VM and that it won't work if the password doesn't comply.

Reply:
Fixed

Reply:
:+1: